### DISCUSSION TITLE
Testing on OLCF Spock (pre-Frontier)

### DISCUSSION THREAD

**Discussion by asedova** [cp2k/dbcsr#473] [Q&A]
Hey all! I've just started trying to test DBCSR on the AMD GPU machine at OLCF (the dev version of Frontier). I was wondering who I can contact repeatedly about this, if possible  :). I see that Shoshana is now no longer at CSCS (she was really helpful on Summit and found a workaround for a terrible and strange issue we had there that is still not really understood but prevented CP2K from running on Summit for a long time). I have a weird error building with the examples turned on in the cmake, but not otherwise, so far.  It would be great to try different combinations of compilers and LA libraries. Would be great to get some feedback and help with possibly hacking the cmake or knowledge of different build options. Cheers-- Ada

**Comment by alazzaro** [comment]
Hello @asedova ! Concerning the DBCSR porting to AMD gpus, there is an ongoing effort mainly driven by @gsitaram (AMD) and @mtaillefumier (CSCS). The activity is for the European pre-exascale system LUMI. Of course, the target is CP2K.
Please, post here any specific request (or open github issues).

**Reply by asedova** [reply]
hey @gsitaram and @mtaillefumier! I am trying to build CP2K and DBCSR on Spock and I have a few errors. With cray programming environment, I am getting 
`In file included from /autofs/nccs-svm1_home2/adaa/CP2K_spock_7_21/cp2k/src/grid/grid_miniapp.c:12:
In file included from /autofs/nccs-svm1_home2/adaa/CP2K_spock_7_21/cp2k/src/grid/../offload/offload_library.h:42:
/sw/spock/spack-envs/views/rocm-4.1.0/hip/include/hip/hip_runtime_api.h:387:2: error: ("Must define exactly one of __HIP_PLATFORM_HCC__ or __HIP_PLATFORM_NVCC__");
#error("Must define exactly one of __HIP_PLATFORM_HCC__ or __HIP_PLATFORM_NVCC__");
 ^
1 error generated.`

With GNU programming environment, I am getting this:

```
 /autofs/nccs-svm1_home2/adaa/CP2K_spock_7_21/cp2k/exts/dbcsr/src/mpi/dbcsr_mpiwrap.F:5391:47:

 5391 |          CALL MPI_FILE_READ_AT_ALL(fh, offset, msg, msg_len, ${mpi_type1}$, MPI_STATUS_IGNORE, ierr)
      |                                               1
......
 5413 |       CALL MPI_FILE_READ_AT_ALL(fh, offset, msg, 1, ${mpi_type1}$, MPI_STATUS_IGNORE, ierr)
      |                                            2   
Error: Rank mismatch between actual argument at (1) and actual argument at (2) (scalar and rank-1)
/autofs/nccs-svm1_home2/adaa/CP2K_spock_7_21/cp2k/exts/dbcsr/src/mpi/dbcsr_mpiwrap.F:5336:48:

 5336 |          CALL MPI_FILE_WRITE_AT_ALL(fh, offset, msg, msg_len, ${mpi_type1}$, MPI_STATUS_IGNORE, ierr)
      |                                                1
......
 5358 |       CALL MPI_FILE_WRITE_AT_ALL(fh, offset, msg, 1, ${mpi_type1}$, MPI_STATUS_IGNORE, ierr)
      |                                             2   
Error: Rank mismatch between actual argument at (1) and actual argument at (2) (scalar and rank-1)
/autofs/nccs-svm1_home2/adaa/CP2K_spock_7_21/cp2k/exts/dbcsr/src/mpi/dbcsr_mpiwrap.F:5284:44: 
```

--and so on for many similar errors. I get the same error when I try to build just DBCSR with GNU, even with MPI set to off in the flags.

**Comment by alazzaro** [comment]
There is a big of mix of errors...

For the first case, this is a CP2K error message. Refer to the CP2K installation file.
For the second case, this is a GNU 10 compiler error. Just add `-fallow-argument-mismatch`.
Since you are compiling CP2K, I would suggest to use the CP2K googleforum.

**Reply by asedova** [reply]
Thanks Alfio!

**Reply by asedova** [reply]
Alfio-- the `-fallow-argument-mismatch` is still required with gcc 11 on Spock. Is the compiler bug also in 11?

**Reply by alazzaro** [reply]
Oh, sorry, there is a misunderstand here. I've never said it is a bug in the compiler... This is an error raised by the compiler because with GCC 10+ we are not supposed to use the MPI F77 interface anymore in the way we are doing in CP2K (GCC became more restrict on that). So, yes, `-fallow-argument-mismatch` is still required....
